src/main/java/org/jenkinsci/plugins/workflow/libs/SCMSourceRetriever.java

```diff
@@ -157,14 +190,19 @@ static void doRetrieve(String name, boolean changelog, @Nonnull SCM scm, FilePat
                 delegate.checkout(run, lease.path, listener, node.createLauncher(listener));
                 return null;
             });
-            // Cannot add WorkspaceActionImpl to private CpsFlowExecution.flowStartNodeActions; do we care?
-            // Copy sources with relevant files from the checkout:
+            if (libraryPath == null) {
+                libraryPath = ".";
+            } else if (PROHIBITED_DOUBLE_DOT.matcher(libraryPath).matches()) {
+                throw new AbortException("Library path may not contain '..'");
```

Comment:
I intentionally did not use the localized message here, because I think we don't normally localize build log content (since it is not produced in a context where this an active user), but I do not remember for sure if this is the right thing to do.